include/aws/cal/private/der.h

```diff
@@ -13,6 +13,21 @@
 struct aws_der_encoder;
 struct aws_der_decoder;
 
+/*
+ * Note: encoder/decoder only support positive integer representations and usage
+ * of signed integers might lead to unexpected results.
+ * Context: DER spec requires ints to be stored in big endian format with MSB
+ * representing signedness. To disambiguate between negative number and big
+ * positive number, null byte can be added in front of positive number. DER spec
+ * requires representation to be the shortest possible one.
+ * When encoding the integer, the encoder assumes its always positive number and
+ * will add leading null byte if MSB is set.
+ * When decoding the integer, null byte is always stripped.
+ * This works fine when using with functions that work with positive ints only
+ * (ex. openssl's BN_bin2bn), but will break with functions that look at msb for
+ * sign (ex. BN_mpi2bn).
```

Comment:
i was somewhat on a fence on this, since der code is fairly tricky and i wanted to avoid modifying it too much, hence the current approach of just fixing 0 case, but not updating interface to handle negatives correctly.
However, i was reviewing the code and fixing der to support negatives properly would not be that hard. Let me pull that into a separate pr

Reply:
I'll take back what i said. supporting negatives in der encoder/decoder would not be as trivial as I initially thought. 

For context, der requires integers to be encoded in big-endian 2s complement format. and to disambiguate between positive and negative number, if the first bit of number is 1, then 0 byte should be prepended. 

The way our current encoder impl works is by checking first bit and prepending 0 byte if the bit is 1. which effectively means that if the negative number is passed in, we encode it into der as positive. 
Similarly decoder always strips first byte if its 0. This has several problems - if integer itself is zero then decoder gets into bad state (problem this pr is addressing), spec says representation should be minimal, but there are a few libs out there that do not follow that part exactly and hence most decoders tend to be lenient on the amount of prepended 0s (arguably very minor, but something to consider for future). 

I see 2 ways to address handling of negatives:
- push 0 padding to calling code and then der enc/dec can remove special handling around first bit. would require some refactoring at call sites and dec/enc can remove extra leading bit logic. prepending at the call site will also require allocation in some cases. It can be argued both ways whether dec enc/dec should own the logic 0 padding.
- introduce another method for writing/reading signed ints (i prefer this approach more). will require some refactoring of internals.

neither of the above approaches fix 0 case by themselves and we would still need 0 fix similar to this pr. Both approaches will require some refactoring to support negatives and truthfully since we dont have a usecase for negatives right now, i dont think its worth adding support for negatives.

Reply:
yeah, let me remove that. i'll keep the change in der impl for now, so tests work. and will remove it later.